core/string/valid_encoding_spec.rb

```diff
@@ -98,6 +98,9 @@
     str.force_encoding('MacJapanese').valid_encoding?.should be_false
     str.force_encoding('UTF-7').valid_encoding?.should be_true
     str.force_encoding('UTF8-MAC').valid_encoding?.should be_true
+    ruby_version_is "2.7" do
+      str.force_encoding('CESU-8').valid_encoding?.should be_true
```

Comment:
Could you move this in a separate `it`?
The reason is otherwise the entire spec example would fail on some Ruby implementation (reporting >= 2.7) which didn't add CESU-8 yet.
In general, `ruby_version_is` should be outside `it` blocks.

Reply:
Maybe a better place to test this would be the Encoding.list spec or so.
Testing specific characteristics of [CESU-8](https://en.wikipedia.org/wiki/CESU-8) (e.g., how some codepoint is encoded) would be even better.

Reply:
@eregon Thanks for review!

> Could you move this in a separate it?
Sure!

> Maybe a better place to test this would be the Encoding.list spec or so

As I see now there no tests for specific encodings in `core/encoding/list_spec.rb`. Maybe we should add something like that?
```ruby
it 'includes CESU-8' do
  Encoding.list.include?(Encoding::CESU_8)
end

it 'includes UTF-8' do
  Encoding.list.include?(Encoding::UTF_8)
end
...
# and so on
```

>Testing specific characteristics of CESU-8 (e.g., how some codepoint is encoded) would be even better.
Something like in [Rust docs](https://docs.rs/cesu8/1.1.0/cesu8/)?

Reply:
>  Maybe we should add something like that?

Feel free to add UTF-8 too, but I wouldn't add all the others.
CESU-8 is special in that it's a newly-added Encoding.

Yes, for instance testing `10400.chr(Encoding::CESU_8).bytes.should == [...]` (from the Wikipedia example).

Reply:
@eregon I have added more specific specs. Can you check please?